promgen/views.py

```diff
@@ -1374,3 +1376,31 @@ def post(self, request, pk):
         return JsonResponse(
             {request.POST["target"]: render_to_string("promgen/ajax_clause_check.html", result)}
         )
+
+
+class PromqlQuery(View):
+    def get(self, request):
+        if not all(x in request.GET for x in ["shard", "query"]):
+            return HttpResponse("BAD REQUEST", status=400)
+
+        shard = models.Shard.objects.get(pk=request.GET["shard"])
+        params = {"query": request.GET["query"]}
+        headers = {}
+
+        if shard.basic_auth:
+            user_pass = b64encode(
+                f"{shard.basic_auth_user}:{shard.basic_auth_password}".encode()
+            ).decode()
+            headers["Authorization"] = f"Basic {user_pass}"
```

Comment:
```suggestion
        if shard.basic_auth_user:
            user_pass = b64encode(
                f"{shard.basic_auth_user}:{shard.basic_auth_password}".encode()
            ).decode()
            headers["Authorization"] = f"Basic {user_pass}"
```

Python would like you do a check like that, so having an extra `basic_auth` field seems a little unnecessary, since it's used nowhere else, and not even used in the `admin.py` as a kind of filter 🤔